internal/provider/opts.go

```diff
@@ -103,6 +103,9 @@ type Opts struct {
 	// UserMode is whether to use the user's systemd or the system's.
 	UserMode bool
 
+	// MapRoot maps the root user to this UID (defaults to 0).
+	MapRoot int
```

Comment:
Given it's an int, we should ensure its validation in `SetDefaults`. Can you please add such validation?

Reply:
to check for <0? You also want SetDefaults to return an error in that case?
Note user.LookupId with a negative number will fail

Reply:
Either return an error or override to zero, but I guess an error is good to convey a misconfiguration issue to the user.

Reply:
Ensuring the ID actually exists also makes sense to me if that's what you're proposing.